Extensions/SpatialSound/JsExtension.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+/**
+ * This is a declaration of an extension for GDevelop 5.
+ *
+ * ℹ️ Changes in this file are watched and automatically imported if the editor
+ * is running. You can also manually run `node import-GDJS-Runtime.js` (in newIDE/app/scripts).
+ *
+ * The file must be named "JsExtension.js", otherwise GDevelop won't load it.
+ * ⚠️ If you make a change and the extension is not loaded, open the developer console
+ * and search for any errors.
+ *
+ * More information on https://github.com/4ian/GDevelop/blob/master/newIDE/README-extensions.md
+ */
+
+/*::
+// Import types to allow Flow to do static type checking on this file.
+// Extensions declaration are typed using Flow (like the editor), but the files
+// for the game engine are checked with TypeScript annotations.
+import { type ObjectsRenderingService, type ObjectsEditorService } from '../JsExtensionTypes.flow.js'
+*/
+
+module.exports = {
+  createExtension: function (
+    _ /*: (string) => string */,
+    gd /*: libGDevelop */
+  ) {
+    const extension = new gd.PlatformExtension();
+    extension.setExtensionInformation(
+      'SpatialSound',
+      _('Spatial sound'),
+      _(
+        'Allow positioning sounds in 3D space, using a stereo system to make the player know where a sound came from.'
+      ),
+      'Arthur Pacaud (arthuro555)',
+      'MIT'
+    );
+
+    extension
+      .addAction(
+        'SetSoundPosition',
+        _('Set position of sound'),
+        _('Sets the spatial position of a sound.'),
```

Comment:
Can we write in the description something about the units of the position? Like if I'm at pos 0,0,0 and the sounds is at 2,0,0, will I hear the sound? Basically, "how far" is "1"? :)

Reply:
I am not sure to be honest. All I could find about WebAudio spatialization is [this resource](https://www.w3.org/TR/webaudio/#Spatialization-panning-algorithm), and it doesn't make it very clear at which point you wouldn't hear anymore (unless I overread something)

Reply:
See this interesting article: https://medium.com/@kfarr/understanding-web-audio-api-positional-audio-distance-models-for-webxr-e77998afcdff

The attenuation follows an *inverse distance model*, meaning that by default the sound will be reduce by two at a distance of 2, and divided by 2 again at a distance of 4.